sdk/python/lib/pulumi/runtime/rpc_manager.py

```diff
@@ -65,11 +65,13 @@ def do_rpc(self, name: str, rpc_function: Callable[..., Awaitable[None]]) -> Cal
                 log.debug(f"recorded new RPC, {self.count} RPCs outstanding")
 
             try:
-                await rpc_function(*args, **kwargs)
+                result = await rpc_function(*args, **kwargs)
```

Comment:
The result is saved here because `invoke` returns a value that we want to bubble up to the caller.